### PR TITLE
Fix locker physical interaction check

### LINKED ISSUE #1517: GoI rep doesn't have access to his locker
Title

### PR DESCRIPTION
**Pull request by checkraisefold**
## About the Pull Request

closes #1517 

## Why It's Good For The Game

actually fix lockercode

## Changelog

:cl:
fix: Personal lockers can now only be opened by someone with the same or higher access as the person who owns them!
fix: Lockers will actually say 'Access Denied' when you have invalid access to use them.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm [hunk 1]
```diff
@@ -44,8 +43,11 @@
 		if(locked)
 			id_card = istype(id_card) ? id_card : user.GetIdCard()
 			if (id_card)
+				var/list/id_acc = id_card.GetAccess()
+				req_access = id_acc.Copy()
```
Comment (TheDarkElites) [inline code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: You can literally just also add ACCESS_ALL_PERSONAL_LOCKERS here to solve the issue of ACCESS_ALL_PERSONAL_LOCKERS not working

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm (hunk 1 above)
Comment (checkraisefold) [reply code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: I tried that except then you won't be able to open it UNLESS you have ACCESS_ALL_PERSONAL_LOCKERS 
Unless theres a way to have optional access

Comment (TheTitanicEngineer) [reply code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: req_access works the same as doors iirc, this mean that you will need ALL the accesses listed under req_access.
To work around this I believe you can also use req_one_access

**Comment by Tsurupeta** [comment]
The "personal locker" name fully describes its function. It's a personal locker. If a class-D (example) locks his shit in personal locker, with your new solution any other class-D will be able to unlock it. Putting GOI rep's items into locker that's designed for storing personal items, i.e it's misuse, is the issue here.

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm (hunk 1 above)
Comment (TheDarkElites) [reply code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: Right forgot about that, though tsurpta pointed out another issue

**Comment by checkraisefold** [comment]
Tested and working.

**Review by Tsurupeta — APPROVED** [review]

**Review by TheDarkElites — APPROVED** [review]

**Pull request merged**